### PR TITLE
rockcraft file

### PR DESCRIPTION
**Pull request by maci3jka**
KU-1064

### CODE REVIEW

File: rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
+name: fluent-bit
+summary: ROCK for the fluent-bit Project.
+description: This ROCK is a drop in replacement for the fluent/fluent-bit image.
+version: "2.1.6"
+license: Apache-2.0
+
+base: bare
+build-base: ubuntu@22.04
+#base: ubuntu@22.04
```
Comment (bschimke95) [inline rockcraft.yaml]: obsolete comment

File: rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      ${CRAFT_PART_SRC}/conf/plugins.conf \
+      ${CRAFT_PART_INSTALL}/fluent-bit/etc/
+      
+      # Generate schema and include as part of the container image
+      ${CRAFT_PART_INSTALL}/fluent-bit/bin/fluent-bit -J > ${CRAFT_PART_INSTALL}/fluent-bit/etc/schema.json
+      
+      # Get missing packages 
+      wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.0g-2ubuntu4_amd64.deb
+      dpkg -x ./libssl1.1_1.1.0g-2ubuntu4_amd64.deb ${CRAFT_PART_INSTALL}/
+      wget http://http.us.debian.org/debian/pool/main/o/openldap/libldap-2.4-2_2.4.47+dfsg-3+deb10u7_amd64.deb
```
Comment (bschimke95) [inline rockcraft.yaml]: can we get `openldap` from ubuntu instead?

File: tests/integration/conftest.py
Comment (bschimke95) [inline tests/integration/conftest.py:1]: please remove all of this and use the https://github.com/canonical/k8s-test-harness now that we have it.

**Review by bschimke95 — COMMENTED** [review]
first iteration, great work so far

File: rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+services:
+  fluent-bit:
+    override: replace
+    summary: "fluent-bit service"
+    startup: enabled
+    command: "/fluent-bit/bin/fluent-bit [ -c /fluent-bit/etc/fluent-bit.conf ]"
+    on-failure: shutdown
+
+parts:
+  cmake:
```
Comment (berkayoz) [inline rockcraft.yaml]:
```suggestion
  fluent-bit:
```
I think we are building fluent-bit here, so the part name of `cmake` does not make too much sense.

File: tests/templates/cert-manager-test.yaml
Comment (berkayoz) [inline tests/templates/cert-manager-test.yaml:1]: Should this be removed? Do we need this here?

File: tests/integration/test_fluent_bit.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "k8s",
+            "kubectl",
+            "rollout",
+            "status",
+            "daemonset",
+            "fluent-bit",
+            "--namespace",
+            "fluent-bit",
+            "--timeout",
+            "180s",
```
Comment (berkayoz) [inline tests/integration/test_fluent_bit.py]: Does this verify if pods are ready or should we do something like
```
"k8s",
"kubectl",
"wait",
"--for=condition=ready",
"pod",
"-l",
"<pod-label>",
"--timeout",
"180s",
```
to wait for pods to be ready

**Review by berkayoz — COMMENTED** [review]
Great work, left some comments and questions

File: tests/integration/test_fluent_bit.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        "--set",
+        "securityContext.runAsUser=584792",
+    ]
+
+    module_instance.exec(helm_command)
+
+    exec_util.stubbornly(retries=5, delay_s=5).on(module_instance).exec(
+        [
+            "k8s",
+             "kubectl",
```
Comment (claudiubelu) [inline tests/integration/test_fluent_bit.py]: you have a lint error here btw. This line has one extra space.

**Review by berkayoz — APPROVED** [review]
LGTM

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**